src/esrs/aggregate.rs

```diff
@@ -85,57 +84,32 @@ pub trait AggregateManager: Aggregate {
         let events: Vec<Self::Event> = <Self as Aggregate>::handle_command(aggregate_state.inner(), command)?;
         let stored_events: Vec<StoreEvent<Self::Event>> = self.store_events(&mut aggregate_state, events).await?;
 
-        Ok(<Self as AggregateManager>::apply_events(aggregate_state, stored_events))
-    }
-
-    /// Responsible for applying events in order onto the aggregate state, and incrementing the sequence number.
-    ///
-    /// `events` will be passed in order of ascending sequence number.
-    ///
-    /// You should _avoid_ implementing this function, and be _very_ careful if you decide to do so.
-    fn apply_events(
-        aggregate_state: AggregateState<Self::State>,
-        store_events: Vec<StoreEvent<Self::Event>>,
-    ) -> AggregateState<Self::State> {
-        let sequence_number: SequenceNumber = store_events.last().map_or(0, StoreEvent::sequence_number);
-
-        let events: Vec<Self::Event> = store_events
-            .into_iter()
-            .map(|store_event| store_event.payload)
-            .collect();
-
-        let inner: Self::State = <Self as Aggregate>::apply_events(aggregate_state.inner, events);
-
-        AggregateState {
-            sequence_number,
-            inner,
-            ..aggregate_state
-        }
+        Ok(aggregate_state.apply_store_events(stored_events, Self::apply_events))
     }
 
     /// Loads an aggregate instance from the event store, by applying previously persisted events onto
     /// the aggregate state by order of their sequence number
     ///
     /// You should _avoid_ implementing this function, and be _very_ careful if you decide to do so.
-    async fn load(&self, aggregate_id: impl Into<Uuid> + Send) -> Option<AggregateState<Self::State>> {
+    async fn load(
+        &self,
+        aggregate_id: impl Into<Uuid> + Send,
+    ) -> Result<Option<AggregateState<Self::State>>, Self::Error> {
```

Comment:
Exposing the errors, rather than randomly swallowing it into the option.
We had to debug this at least twice in our app, and it was SO frustrating.